### PR TITLE
Fixed some TypeScript Lint warnings.

### LINKED ISSUE #1201: Fix Typescript Rules lint warnings
**Describe the bug**
Currently, there are too many **Typescript Rules** ESLint warnings in the **develop** branch. 

Most of them are:-
"@typescript-eslint/ban-ts-comment",
"@typescript-eslint/no-explicit-any",
"@typescript-eslint/no-inferrable-types",
"@typescript-eslint/no-non-null-asserted-optional-chain",
"@typescript-eslint/no-non-null-assertion",
"@typescript-eslint/no-var-requires",

**To Reproduce**

1. Clone the **talawa-api** repository.
2. Install the project dependencies. (npm i)
3. In .eslintrc.json, change **Typescript Rule**, from line 49 to 54, from "off" to "warn", just like this:-
pic2

4. Run **npm run lint:check**

**Expected behavior**
Lint should report 0 Warnings and 0 Errors

**Actual behavior**
Lint now reports too many warnings.

**Additional details**
Multiple contributors can work together as there are too many warnings.

### PR DESCRIPTION
**Pull request by git-init-priyanshu**
**What kind of change does this PR introduce?**
This pull request addresses some of the TypeScript lint warnings.

**Issue Number:**
#1201 

Fixes #1201
 

**Did you add tests for your changes?**
 No. I have tested it against the existing test cases.

**Snapshots/Videos:**
N/A

**If relevant, did you update the documentation?**
No

**Summary**
This PR fixes the following lint warnings:

1. ban-ts-comment
2. no-explicit-any
3. no-inferrable-types
4. no-non-null-asserted-optional-chain
5. no-non-null-assertion
6. no-var-requires

**Does this PR introduce a breaking change?**
No

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**
Yes

### CODE REVIEW

**Comment by git-init-priyanshu** [comment]
Hey @tasneemkoushar,
Can you please review my PR. Thanks.

**Comment by git-init-priyanshu** [comment]
Hey @palisadoes can you review my PR please. Thanks.

**Comment by palisadoes** [comment]
@tasneemkoushar Please review

**Comment by palisadoes** [comment]
@tasneemkoushar Please review

**Comment by palisadoes** [comment]
@git-init-priyanshu  Shouldn't you also be editing the  `.eslintrc.json` file too? Ideally we should make the tests fail when the linting fails, this will help a lot with our code quality. I've added @EshaanAgg to provide his feedback.

File: src/resolvers/Mutation/createDirectChat.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export const createDirectChat: MutationResolvers["createDirectChat"] = async (
       _id: args.data.organizationId,
     }).lean();
 
-    await cacheOrganizations([organization!]);
+    if (organization) await cacheOrganizations([organization]);
```
Comment (EshaanAgg) [inline src/resolvers/Mutation/createDirectChat.ts:51]: You have added this `if (organization)` check in all of the caching statements to remove the use of the `!` operator. Still, the same is unnecessary as it is guaranteed to exist, and thus the use of the `!` operator. This change adds unnecessary verbosity to the code.
We should revert this change or find a more elegant way to deal with the same. What do you think @kb-0311?

**Review by EshaanAgg — CHANGES_REQUESTED** [review]
Please also add all these rules to the linting config files so that husky and the workflows catch them.

File: src/resolvers/Mutation/createDirectChat.ts (hunk 1 above)
Comment (git-init-priyanshu) [reply src/resolvers/Mutation/createDirectChat.ts:51]: Ok got it. I think I need to find a better way to remove the `!` operator

**Comment by palisadoes** [comment]
Please fix the conflicting file

**Comment by tasneemkoushar** [comment]
@git-init-priyanshu can you resolve the conflicts?

**Comment by noman2002** [comment]
@git-init-priyanshu Please fix the conflicting files.

**Comment by git-init-priyanshu** [comment]
I apologize for the inactivity. I have semester exams starting next week, so I won't have the time to work on this. Could you please grant me more time? Thank you.

**Comment by tasneemkoushar** [comment]
@git-init-priyanshu you can do it after your exam.  FYI some linting issues have been fixed by @parthiv360 . You can check that here in this [PR](https://github.com/PalisadoesFoundation/talawa-api/pull/1412). Please resolve conflicting files and try to fix linting issues which aren't covered in PR #1412 once your exam finishes.

**Comment by git-init-priyanshu** [comment]
> @git-init-priyanshu you can do it after your exam.  FYI some linting issues have been fixed by @parthiv360 . You can check that here in this [PR](https://github.com/PalisadoesFoundation/talawa-api/pull/1412). Please resolve conflicting files and try to fix linting issues which aren't covered in PR #1412 once your exam finishes.

Yeah, sure.

**Comment by noman2002** [comment]
@git-init-priyanshu Please fix the conflicting files.

**Comment by tasneemkoushar** [comment]
@git-init-priyanshu any update on this PR

**Comment by git-init-priyanshu** [comment]
> @git-init-priyanshu any update on this PR

My exams just finished today, so I will be working on this now.

**Comment by git-init-priyanshu** [comment]
@tasneemkoushar I have made the changes. Can you check now.

File: tests/utilities/imageAlreadyInDbCheck.spec.ts [hunk 2]
```diff
@@ -190,7 +191,8 @@ describe("utilities -> imageAlreadyInDbCheck", () => {
 
     try {
       await imageAlreadyInDbCheck(null, testNewImagePath);
-    } catch (error: any) {
+    } catch (error: unknown) {
+      if (!(error instanceof ApplicationError)) return;
```
Comment (EshaanAgg) [inline tests/utilities/imageAlreadyInDbCheck.spec.ts:195]: Make the test fail in such a case.

File: tests/utilities/encodedImageStorage/uploadEncodedImage.spec.ts [hunk 3]
```diff
@@ -32,7 +33,8 @@ describe("src -> utilities -> encodedImageStorage -> uploadEncodedImage", () =>
         "NAAAAKElEQVQ4jWNgYGD4Twzu6FhFFGYYNXDUwGFpIAk2E4dHDRw1cDgaCAASFOffhEIO" +
         "3gAAAABJRU5ErkJggg==";
       await uploadEncodedImage(img, null);
-    } catch (error: any) {
+    } catch (error: unknown) {
+      if (!(error instanceof ApplicationError)) return;
```
Comment (EshaanAgg) [inline tests/utilities/encodedImageStorage/uploadEncodedImage.spec.ts:37]: Make the test fail here instead of returning.

File: tests/utilities/adminCheck.spec.ts [hunk 4]
```diff
@@ -50,7 +50,8 @@ describe("utilities -> adminCheck", () => {
         testUser?._id,
         testOrganization ?? ({} as InterfaceOrganization)
       );
-    } catch (error: any) {
+    } catch (error: unknown) {
+      if (!(error instanceof Error)) return;
```
Comment (EshaanAgg) [inline tests/utilities/adminCheck.spec.ts:54]: Ditto.

**Review by EshaanAgg — CHANGES_REQUESTED** [review]
Needs minor changes.

File: tests/utilities/imageAlreadyInDbCheck.spec.ts (hunk 2 above)
Comment (git-init-priyanshu) [reply tests/utilities/imageAlreadyInDbCheck.spec.ts:195]: > Make the test fail in such a case.

if (!(error instanceof Error)) throw new Error;

Sorry for asking but will it work?

Comment (EshaanAgg) [reply tests/utilities/imageAlreadyInDbCheck.spec.ts:195]: This is not idiomatic. 
Use [Vitest error assertions](https://vitest.dev/api/expect.html#tothrowerror) instead.

File: src/resolvers/Mutation/createDirectChat.ts (hunk 1 above)
Comment (git-init-priyanshu) [reply src/resolvers/Mutation/createDirectChat.ts]: Hey, for this can we do something like this?: we can check if we are getting `id` from the `args` and if not we can throw an error. So this will make sure that that the ids we are passing to the `findOrganizationsInCache` function will never be null.

**Comment by git-init-priyanshu** [comment]
Hey @EshaanAgg can please check now.

**Comment by EshaanAgg** [comment]
Please fix the failing workflows.

**Comment by git-init-priyanshu** [comment]
> Please fix the failing workflows.

can you check now?

**Comment by EshaanAgg** [comment]
@git-init-priyanshu The tests are still failing.

**Review by EshaanAgg — CHANGES_REQUESTED** [review]
The PR has the right spirit but needs work. Instead of trying to fix the lining warnings, please try to understand the intent behind using `!` or `?` and then write the resolvers accordingly. TypeScript and lining rules should help to enforce these patterns, not bother.

**Comment by palisadoes** [comment]
@git-init-priyanshu

1. You have made almost 40 commits to this PR without closing it. 
2. Something is wrong with your methodology or the original issue needs to be adjusted. We have a linting issue created to just check on the deltas which may be more appropriate.
    - https://github.com/PalisadoesFoundation/talawa-api/issues/1548

Please close this PR and the issue if the issue above is a more suitable solution.

**Comment by git-init-priyanshu** [comment]
I totally agree with you, sir. I was also losing motivation to work on this PR. I will review my methodology and make sure to work on this. If you have any specific suggestions or insights, I would welcome them to ensure that I can improve.

**Comment by palisadoes** [comment]
1. Thanks. I'll close this and the issue too.
2. Focus on other issues that will reactivate your motivation.
